### PR TITLE
dev UX fixups

### PR DESCRIPTION
**Pull request by ewdurbin**
- A couple missed `docker-compose` -> `docker compose`
- stripe-mock now has arm builds! https://github.com/stripe/stripe-mock/issues/281
- Build/run our docs stuff in their own image
- Remove some stray debugging that was causing dev-docs to open a python REPL 🙃
- Better mechanism for sharing container images between services
- Don't delete compose volumes on `make stop`, rather do that only when `make purge` is called

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -186,22 +186,33 @@ services:
...
 
   blog:
-    image: warehouse:docker-compose
+    image: warehouse:docker-compose-docs
+    pull_policy: never
     command: mkdocs serve -a 0.0.0.0:8000 -f docs/blog.yml
     volumes:
-      - ./docs/blog.yml:/opt/warehouse/src/docs/blog.yml:z
-      - ./docs/blog:/opt/warehouse/src/docs/blog:z
+      - ./.git:/opt/warehouse/src/.git:z
```
Comment (miketheman) [inline docker-compose.yml:214]: I had a similar note about git directory previously 
https://github.com/pypi/warehouse/pull/13231#discussion_r1142476044

If it needs to be mounted, should it be in ro mode?

File: Dockerfile [hunk 2]
```diff
@@ -132,7 +187,7 @@ RUN set -x \
     && apt-get update \
     && apt-get install --no-install-recommends -y \
         libpq5 libxml2 libxslt1.1 libcurl4  \
-        $(if [ "$DEVEL" = "yes" ]; then echo 'bash libjpeg62 postgresql-client build-essential libffi-dev libxml2-dev libxslt-dev libpq-dev libcurl4-openssl-dev libssl-dev git libcairo2-dev libfreetype6-dev libjpeg-dev libpng-dev libz-dev'; fi) \
```
Comment (miketheman) [inline Dockerfile:135]: When the blog was added, this is the list of deps added for it to work:
git libcairo2-dev libfreetype6-dev libjpeg-dev libpng-dev libz-dev

I think all of them should migrate to the new docs target instead, right?

File: docker-compose.yml (hunk 1 above)
Comment (ewdurbin) [reply docker-compose.yml:214]: oh heck yeah, good call. it's just for generating the timestamps on blogs. i missed it getting removed.

File: Dockerfile (hunk 2 above)
Comment (ewdurbin) [reply Dockerfile:135]: ty.

File: docker-compose.yml [hunk 3]
```diff
@@ -186,22 +186,33 @@ services:
...
 
   blog:
-    image: warehouse:docker-compose
+    image: warehouse:docker-compose-docs
+    pull_policy: never
     command: mkdocs serve -a 0.0.0.0:8000 -f docs/blog.yml
     volumes:
-      - ./docs/blog.yml:/opt/warehouse/src/docs/blog.yml:z
-      - ./docs/blog:/opt/warehouse/src/docs/blog:z
+      - ./.git:/opt/warehouse/src/.git:ro
```
Comment (ewdurbin) [inline docker-compose.yml:214]: This is needed for rss feedz

**Review by miketheman — APPROVED** [review]

**Pull request merged**